tests/main/retry-network/task.yaml

```diff
@@ -34,7 +43,7 @@ execute: |
     MATCH "(Temporary failure in name resolution|network is unreachable)" < stderr
 
     echo "Now without DNS resolver"
-    UBUNTU_IP="$(getent hosts www.ubuntu.com|awk '{print $1 }' | head -n1)"
+    UBUNTU_IP="$(nslookup www.ubuntu.com | grep 'Server:' | awk '{print $2 }')"
```

Comment:
Hmm, with my machine, nslookup provides the Server as the DNS resolver, which is localhost:

```
$ nslookup www.ubuntu.com
Server:		127.0.0.53
Address:	127.0.0.53#53

Non-authoritative answer:
www.ubuntu.com	canonical name = website-content-cache.canonical.com.
Name:	website-content-cache.canonical.com
Address: 91.189.91.44
Name:	website-content-cache.canonical.com
Address: 91.189.88.181
Name:	website-content-cache.canonical.com
Address: 91.189.88.180
Name:	website-content-cache.canonical.com
Address: 91.189.91.45
Name:	website-content-cache.canonical.com
Address: 2001:67c:1360:8001::2c
Name:	website-content-cache.canonical.com
Address: 2001:67c:1360:8001::2b
Name:	website-content-cache.canonical.com
Address: 2001:67c:1562::20
Name:	website-content-cache.canonical.com
Address: 2001:67c:1562::1f
```
which isn't what we want for this test IIUC. `getent` seems to support a v4 specific hosts command with `getent ahostsv4`, with a slightly different output format. I think we could use something like:

```
$ getent ahostsv4 www.ubuntu.com | awk '{print $1}' | head -n1
91.189.91.44
```

I checked and this getent command works the same on 14.04 as it does on 20.04, so I'm reasonably certain that this form would work on other distros as well.

Reply:
Thanks for the review, I already pushed the fix you suggested